--- libvirt-1.2.7.orig/src/conf/domain_conf.c	2014-07-31 11:39:53.000000000 -0400
+++ libvirt-1.2.7/src/conf/domain_conf.c	2014-10-14 09:42:59.699712574 -0400
@@ -16071,6 +16071,8 @@ virDomainNetDefFormat(virBufferPtr buf,
                                       def->data.vhostuser->data.nix.path);
                 if (def->data.vhostuser->data.nix.listen)
                     virBufferAddLit(buf, " mode='server'");
+                else
+                    virBufferAddLit(buf, " mode='client'");
                 virBufferAddLit(buf, "/>\n");
             }
             break;
