commit e4deaf5d878f34852e31746588c1946f57b86472
Author: Michal Dubiel <md@semihalf.com>
Date:   Wed Mar 25 12:04:21 2015 -0400

    Update version

diff --git a/debian/changelog b/debian/changelog
index f36983a..ae76e2f 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,3 +1,9 @@
+libvirt-python (1.2.12-0ubuntu1-contrail1) precise; urgency=medium
+
+  * Initial back port for Ubuntu 12.04.
+
+ -- Michal Dubiel <md@semihalf.com>  Wed, 25 Mar 2015 11:05:55 +0100
+
 libvirt-python (1.2.12-0ubuntu1) vivid; urgency=medium
 
   * New uptream release.
