diff -Nuar libvirt-2.5.0.orig/debian/changelog libvirt-2.5.0/debian/changelog
--- libvirt-2.5.0.orig/debian/changelog	2017-08-02 19:07:52.004000000 -0700
+++ libvirt-2.5.0/debian/changelog	2017-08-02 19:15:21.356000000 -0700
@@ -1,3 +1,22 @@
+libvirt (2.5.0-3ubuntu5.3~cloud0~xenial0+contrail2) xenial-ocata; urgency=medium
+
+  * Support for openstack ocata   
+
+ -- Jeya ganesh babu J <jjeya@juniper.net>  Wed, 02 Aug 2017 20:59:00 -0800
+
+libvirt (2.5.0-3ubuntu5.3~cloud0~xenial0+contrail1) xenial-ocata; urgency=medium
+
+  * Support for openstack newton 
+
+ -- Kiran KN <kirankn@juniper.net>  Sun, 21 May 2017 15:11:17 +0200
+
+libvirt (2.5.0-3ubuntu5.3~cloud0~xenial0+contrail0) xenial-ocata; urgency=medium
+
+  * Support multiqueues with ethernet type interfaces.
+  * Apparmor permissions adjustments.
+
+ -- Michal Dubiel <md@semihalf.com>  Tue, 14 Jun 2016 15:11:17 +0200
+
 libvirt (2.5.0-3ubuntu5.3~cloud0) xenial-ocata; urgency=medium
 
   * New update for the Ubuntu Cloud Archive.
