commit d477fbd19baa67468ed2aaf7b96934b8a5b6cfd1
Author: Michal Dubiel <md@semihalf.com>
Date:   Thu Apr 21 15:33:32 2016 +0200

    Add contrail versioning

diff --git a/debian/changelog b/debian/changelog
index da45ecd..398e06f 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,3 +1,9 @@
+libvirt (1.2.16-2ubuntu11.15.10.4~cloud0+contrailexp1) UNRELEASED; urgency=medium
+
+  * Contrail patches for multiqueue support with ethernet type interfaces
+
+ -- Michal Dubiel <md@semihalf.com>  Thu, 21 Apr 2016 15:31:41 +0200
+
 libvirt (1.2.16-2ubuntu11.15.10.4~cloud0) trusty-liberty; urgency=medium
 
   * New update for the Ubuntu Cloud Archive.
