diff -upr libvirt-1.2.12.orig/debian/apparmor/libvirt-qemu libvirt-1.2.12/debian/apparmor/libvirt-qemu
--- libvirt-1.2.12.orig/debian/apparmor/libvirt-qemu	2015-02-12 21:05:12.000000000 +0100
+++ libvirt-1.2.12/debian/apparmor/libvirt-qemu	2015-12-23 13:06:48.888162789 +0100
@@ -53,6 +53,9 @@
   # unless you absolutely need it.
   deny capability kill,
 
+  # For lttng access.
+  /{dev,run}/shm/lttng-ust-wait-5 r,
+
   # Uncomment the following if you need access to /dev/fb*
   #/dev/fb* rw,
 
@@ -143,6 +146,10 @@
 
   # for access to hugepages
   owner "/run/hugepages/kvm/libvirt/qemu/**" rw,
+  owner "/hugepages/libvirt/qemu/**" rw,
+
+  # for cpu and numa config
+  /sys/devices/system/** r,
 
   # for usb access
   /dev/bus/usb/ r,
