diff --git a/debian/changelog b/debian/changelog
index da45ecd..9860818 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,3 +1,15 @@
+libvirt (1.2.16-2ubuntu11.15.10.4~cloud0+contrail2) trusty-liberty; urgency=medium
+
+  * Backport vhost-user multiqueue support
+
+ -- Michal Dubiel <md@semihalf.com>  Tue, 10 May 2016 16:05:27 +0200
+
+libvirt (1.2.16-2ubuntu11.15.10.4~cloud0+contrail1) trusty-liberty; urgency=medium
+
+  * Contrail patches for multiqueue support with ethernet type interfaces
+
+ -- Michal Dubiel <md@semihalf.com>  Thu, 21 Apr 2016 15:31:41 +0200
+
 libvirt (1.2.16-2ubuntu11.15.10.4~cloud0) trusty-liberty; urgency=medium
 
   * New update for the Ubuntu Cloud Archive.
