diff --git a/debian/changelog b/debian/changelog
index 7cad9d9..3a03f3b 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,3 +1,24 @@
+libvirt (2:1.3.1-1ubuntu10.26~cloud0+contrail2) UNRELEASED; urgency=medium
+
+  * Updated libvirt and Bumped up version
+
+ -- Jeya ganesh babu J <jjeya@juniper.net>  Wed, 04 Jun 2019 21:40:10 -0800
+
+libvirt (1.3.1-1ubuntu10.26~cloud0+contrail2) UNRELEASED; urgency=medium
+
+  * debian/libvirt-bin.upstart: update pre-stop section in upstart
+    script to stop the libvirt-guests service instead of the now
+    removed libvirt-stop-guests script. (LP: #1829823)
+
+ -- Matthew Ruffell <matthew.ruffell@canonical.com>  Tue, 21 May 2019 17:51:43 +1200
+     
+libvirt (1.3.1-1ubuntu10.26~cloud0+contrail1) UNRELEASED; urgency=medium
+
+  * Support multiqueues with ethernet type interfaces.
+  * Apparmor permissions adjustments.
+
+ -- Michal Dubiel <md@semihalf.com>  Tue, 14 Jun 2016 15:11:17 +0200
+ 
 libvirt (1.3.1-1ubuntu10.26~cloud0) trusty-mitaka; urgency=medium
 
   * New update for the Ubuntu Cloud Archive.
