commit ddf1a05b35c7ab913535c70a33accf452f857ca6
Author: Michal Dubiel <md@semihalf.com>
Date:   Thu Mar 19 11:49:08 2015 -0400

    control modifications

diff --git a/debian/control b/debian/control
index b1f7fb3..e7a5017 100644
--- a/debian/control
+++ b/debian/control
@@ -30,8 +30,8 @@ Build-Depends: cdbs (>= 0.4.90~),
  libhal-dev [!linux-any],
  libpciaccess-dev,
  module-init-tools [linux-any],
- policykit-1 (>= 0.105-3ubuntu3),
- libpolkit-gobject-1-dev (>= 0.105-3ubuntu3),
+ policykit-1 (>= 0.104-1ubuntu1.1),
+ libpolkit-gobject-1-dev (>= 0.104-1ubuntu1.1),
  libcap-ng-dev [linux-any],
  libnl-3-dev,
  libnl-route-3-dev,
@@ -53,8 +53,8 @@ Build-Depends: cdbs (>= 0.4.90~),
  ebtables,
  iptables,
  qemu-utils,
- libcgmanager-dev,
- xsltproc
+ xsltproc,
+ autopoint
 Build-Conflicts: dpkg-dev (= 1.15.3)
 Homepage: http://libvirt.org
